dashboard/app/controllers/lessons_controller.rb

```diff
@@ -44,8 +44,25 @@ def student_lesson_plan
     @lesson_data = @lesson.summarize_for_student_lesson_plan
   end
 
-  # GET /lessons/1/edit
+  # GET /lessons/1/edit where 1 is the relative position of the lesson in the script
+  def edit_with_lesson_position
+    script = Script.get_from_cache(params[:script_id])
+    @lesson = script.lessons.find do |l|
+      l.has_lesson_plan && l.relative_position == params[:lesson_position].to_i
+    end
+
+    setup_edit
+    render :edit
+  end
+
+  # GET /lessons/1/edit where 1 is the ID of the lesson
   def edit
+    setup_edit
+  end
+
+  # We have two urls you can use to edit a lesson with a lesson plan. This does the
+  # work for both of them to prepare the data for editing
+  def setup_edit
```

Comment:
I believe this method should be private, so that it cannot be accessed as an action on this controller